src/test/java/com/artipie/npm/proxy/NpmProxyTest.java

```diff
@@ -23,42 +23,42 @@
  */
 package com.artipie.npm.proxy;
 
-import com.amihaiemil.eoyaml.Yaml;
-import com.amihaiemil.eoyaml.YamlMapping;
-import com.artipie.asto.Concatenation;
-import com.artipie.asto.memory.InMemoryStorage;
+import com.artipie.asto.Content;
 import com.artipie.npm.proxy.model.NpmAsset;
 import com.artipie.npm.proxy.model.NpmPackage;
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
 import io.vertx.reactivex.core.Vertx;
-import io.vertx.reactivex.core.http.HttpServer;
 import java.io.IOException;
-import java.net.ServerSocket;
 import java.nio.charset.StandardCharsets;
-import java.util.concurrent.CountDownLatch;
 import org.apache.commons.io.IOUtils;
 import org.hamcrest.MatcherAssert;
-import org.hamcrest.core.IsEqual;
-import org.json.JSONException;
+import org.hamcrest.core.IsSame;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
-import org.skyscreamer.jsonassert.JSONAssert;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
 
 /**
  * Test NPM Proxy works.
  *
  * @since 0.1
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
+@ExtendWith(MockitoExtension.class)
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
-final class NpmProxyTest {
+public final class NpmProxyTest {
```

Comment:
@brastak we have a lot of mockito code. Tests would be so much cleaner if we would use fake implementations instead. If you agree we can add a puzzle for that and solve it in another PR.

Reply:
@Vatavuk Please, clarify what you mean. After Storage and Remote client logic was extracted to separated classes, all tests for data values were moved to corresponded Storage or Remote unit tests. So, here we already ensure that in every step we will get the correct values. The goal of this test is to ensure that the right steps will be executed. We consider that NpmProxy class is a white-box and test it behavior. We don't check the returning values, we check that we firstly need to find data in the local cache, then if we found it, we will return it immediate, otherwise we need to get it from remote repo. And this checks are performed with Mockito verify methods. We can remove this test at all and hope that all necessary checks will be done automatically on integration tests (with the same logic we can remove any other test), but I can not imagine how we can rewrite this test shorter.

Reply:
@brastak I'm saying that we can use fake implementation instead of mocks to test the exact same things, verify that methods are triggered. We can hide method verification in a fake implementation. The point is to avoid mock code repetition in tests. 
Something linke:
```java
remote = new NpmRemote.Fake(Maybe.just(expected))

Assert.assertTrue(remote.loadPackageTriggeredTimes(1))
```
There are no mock preparations, the code is much cleaner.

Reply:
@Vatavuk I think, it will not be easier to support it in long term. We are talking about the idea to avoid mocks setup: instead of this we will implement all necessary logic in the Fakes and will control the Fakes behavior by passing params in the constructor. But even now we need to handle:
1) method call counters;
2) different behavior on the first and second call in some cases.
After optimization logic will be started to implementing, we will need to pass more and more parameters to Fakes.
And several more things than Mockito provides out-of-the-box:
- it allows to check order of calls execution, not only to count the calls (I have plans for this feature);
- it allows to validate passing parameters than increase the coverage.
It is possible to implement these features in Fakes, but it looks like that we will do the own mocking framework. And some of these features will require initial setup too.

Reply:
@brastak ok fair enough. There is still an option to use Mockito inside the fake objects, just a thought, but maybe the fakes would be unnecessary complex...